WeakAuras/History.lua

```diff
@@ -0,0 +1,88 @@
+--[[
+This file contains all of the interfaces to the history collection. Indexed by uid of the active display.
+
+History:
+  Table containing historical information about display:
+  source: string indicating where this data came from. Possible values:
+    "user": aura was created by the user via the WeakAuras interface
+    "import": aura was imported via WeakAuras.Import (i.e. pasted string or addon channel transmission)
+    "addon": aura was imported via WeakAuras.RegisterDisplay (i.e. an addon requested the addition)
+  data: snapshot of data as it was at last import, or nil if source == "user", and the user has not chosen to save their changes to history
+  addon: if source == "addon", then this string indicates which addon supplied the last import
+  skippedVersions: record of versions that the user has indicated they don't wish to see update notifications for
+  allowUpdates: if false, then WeakAuras will ignore any addon-sourced update requests
+  lastUpdate: UNIX timestamp of the last time the user completed an import for this display
+--]]
+
+local WeakAuras = WeakAuras
+local history -- history db upvalue
+
+function WeakAuras.LoadHistory(historydb)
+  history = historydb
+  if history.clearOldHistory == nil then
+    history.clearOldHistory = 30
```

Comment:
To me it looks like you store both the history tables, and this history value in the same table, which makes the later iteration "   for uid, hist in pairs(history) do" also pick up this value. That seems unintended.

Reply:
Ah, no that is an artifact of when I moved this stuff to its own file. Thanks for checking.